test/test_basic.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import base64
+import unittest
+import subprocess
+import os
+import time
+import urllib.error
+import urllib.request
+import ssl
+import json
+
+
+dirname = os.path.dirname(__file__)
+projroot = os.path.join(dirname, "..")
```

Comment:
use os.path.dirname() again instead of `..`; you can merge these, dirname is not used anywhere else.